avocado/utils/cpu.py

```diff
@@ -340,3 +348,24 @@ def get_cpufreq_governor():
     except IOError as err:
         logging.error("Unable to get the current governor\n %s", err)
         return ""
+
+
+def get_pid_cpu(pid):
+    """
+    Get the process used cpus.
+    :param pid: process id
+    :type pid: string
+    :return: A list include all cpus the process used
+    :rtype: list
+    """
+    cmd = "ps -o cpuid -L -p %s" % pid
```

Comment:
We're trying to minimize use of external commands to grab informations which are in standard files[1]. Seems this is also an use case where we can get the information com `proc/[pid]/stat` file instead of to depend on external command. Looking to the kernel doc[2], there's information about `which CPU the task is scheduled on`. Is it what you need?

[1] - https://github.com/avocado-framework/avocado/pull/2851
[2] - https://www.kernel.org/doc/Documentation/filesystems/proc.txt (`Table 1-4: Contents of the stat files (as of 2.6.30-rc7)`)

Reply:
ah sure let me check